--- modules/calendar/serv_calendar.c	2017-04-21 21:39:12.000000000 +0500
+++ modules/calendar/serv_calendar.c	2017-10-09 02:48:45.000000000 +0500
@@ -2580,7 +2580,7 @@
 	{
 
 		/* Tell libical to return errors instead of aborting if it gets bad data */
-		icalerror_errors_are_fatal = 0;
+		icalerror_set_errors_are_fatal(0);
 
 		/* Use our own application prefix in tzid's generated from system tzdata */
 		icaltimezone_set_tzid_prefix("/citadel.org/");
--- modules/crypto/serv_crypto.c	2017-04-21 21:39:12.000000000 +0500
+++ modules/crypto/serv_crypto.c	2017-10-09 03:03:59.000000000 +0500
@@ -83,8 +83,10 @@
 	X509_NAME *name = NULL;
 	FILE *fp;
 
+#ifndef OPENSSL_NO_EGD
 	if (!access(EGD_POOL, F_OK))
 		RAND_egd(EGD_POOL);
+#endif
 
 	if (!RAND_status()) {
 		syslog(LOG_CRIT, "PRNG not adequately seeded, won't do SSL/TLS");
